### PR TITLE
Migrate from glide to Go modules and upgrade AWS Go SDK

### PR DESCRIPTION
**Pull request by shengxil**
Derived from https://github.com/aws/aws-xray-daemon/pull/32. Credit to @jasonrichardsmith

*Issue #, if available:*
https://github.com/aws/aws-xray-daemon/issues/31

*Description of changes:*
moved from glide to go mod.

> The Go community is now using Go Modules to handle dependencies. Please consider using that instead of Glide. Glide is now mostly unmaintained.

Upgraded aws go sdk to newest (1.29.29) for EKS OIDC IAM support. Fixed unit tests accordingly

Added docker build and push functionality. credentials still need to be added to the main repository.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module github.com/aws/aws-xray-daemon
+
+go 1.13
+
+require (
+	github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d // indirect
+	github.com/aws/aws-sdk-go v1.29.29
```
Comment (willarmiros) [inline go.mod]: Thanks for updating AWS SDK!

File: pkg/conn/conn_test.go [hunk 2]
```diff
@@ -305,7 +305,7 @@ func TestLoadEnvConfigCreds(t *testing.T) {
 	assert.Equal(t, "", valueA.AccessKeyID, "Expect the value to be empty")
 	assert.Equal(t, "", valueA.SecretAccessKey, "Expect the value to be empty")
 	assert.Equal(t, "", valueA.SessionToken, "Expect the value to be empty")
-	assert.Equal(t, "", valueA.ProviderName, "Expect the value to be empty")
+	assert.Equal(t, "AssumeRoleProvider", valueA.ProviderName, "Expect the value to be empty")
```
Comment (willarmiros) [inline pkg/conn/conn_test.go]: NP: Update the message on this assertion?

File: makefile [hunk 3]
```diff
@@ -77,16 +76,16 @@ package-rpm:
 .PHONY: test
 test:
 	@echo "Testing daemon"
-	go test -cover ${DAEMON_ONLY_PKGS}
+	go test -cover ./...
 
 vet:
-	${GO_VET_CMD} ${SDK_BASE_FOLDERS}
+	${GO_VET_CMD} ./...
```
Comment (willarmiros) [inline makefile]: NP: Consider eliminating `GO_VET_CMD` and replacing with `go vet` to be consistent with other Makefile targets. According to the [docs](https://godoc.org/github.com/golangci/govet#hdr-Other_flags), the `-all` flag is unnecessary.

Comment (shengxil) [reply makefile]: Good catch!

File: pkg/conn/conn_test.go (hunk 2 above)
Comment (shengxil) [reply pkg/conn/conn_test.go]: Yes, will update the message

**Pull request merged**